packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -1014,7 +1014,7 @@ class DeviceDomain extends Domain {
     );
     return <String, Object?>{
       'started': result.started,
-      'observatoryUri': result.observatoryUri?.toString(),
+      'vmServiceUri': result.vmServiceUri?.toString(),
```

Comment:
The IDE plugins using `daemon` don't know about this key, I think you need to pass both for awhile:
https://github.com/flutter/flutter-intellij/blob/10fb2dd347c6553f40d18071bd669c4bf8a7aadb/flutter-idea/src/io/flutter/run/test/FlutterTestEventsConverter.java#L67

Make sure you test the IDEs on this change to ensure nothing else is broken.

Reply:
Checked with Intellij and VSCode, everything seems to be working with the targets I tried.

Reply:
Looks like tests didn't work from IntelliJ: https://github.com/flutter/flutter/pull/121555